merlion/models/forecast/autoregressivebase.py

```diff
@@ -0,0 +1,251 @@
+import logging
+from typing import List, Tuple
+
+import numpy as np
+import pandas as pd
+
+from merlion.models.forecast.base import ForecasterConfig, ForecasterBase
+from merlion.utils.time_series import to_pd_datetime, TimeSeries
+from merlion.models.utils.rolling_window_dataset import RollingWindowDataset, max_feasible_forecast_steps
+
+logger = logging.getLogger(__name__)
+
+
+class AutoRegressiveForecasterConfig(ForecasterConfig):
+    """
+    Configuration class for auto-regressive forecaster model.
+    """
+
+    def __init__(
+        self,
+        maxlags: int,
+        max_forecast_steps: int = None,
+        target_seq_index: int = None,
+        prediction_stride: int = 1,
+        **kwargs,
+    ):
+        """
+        :param maxlags: Max # of lags for forecasting
+        :param max_forecast_steps: Max # of steps we would like to forecast for.
+        :param target_seq_index: The index of the univariate (amongst all
+            univariates in a general multivariate time series) whose value we
+            would like to forecast.
+        :param prediction_stride: the prediction step for training and forecasting
+
+            - If univariate: the sequence target of the length of prediction_stride will be utilized, forecasting will
+              be done autoregressively, with the stride unit of prediction_stride
+            - If multivariate:
+
+                - if = 1: the autoregression with the stride unit of 1
+                - if > 1: only support sequence mode, and the model will set prediction_stride = max_forecast_steps
```

Comment:
Rephrase: "only support directly forecasting the next prediction_stride steps in the future. Autoregression not supported. Note that the model will set prediction_stride = max_forecast_steps"

Reply:
Done